### PR TITLE
Fix combine objects optimizations

### PR DESCRIPTION
**Pull request by a5i**

### CODE REVIEW

File: pdf/model/optimize/optimizer.go [hunk 1]
```diff
@@ -82,3 +82,22 @@ func replaceObjectsInPlace(objects []core.PdfObject, objTo map[core.PdfObject]co
 		}
 	}
 }
+
+// Update all the object numbers prior to get hash of objects.
+func updateObjectNumbers(objects []core.PdfObject) {
+	// Update numbers
+	for idx, obj := range objects {
+		if io, isIndirect := obj.(*core.PdfIndirectObject); isIndirect {
```
Comment (gunnsth) [inline pdf/model/optimize/optimizer.go]: A type switch would be cleaner for this

File: pdf/creator/creator_test.go [hunk 2]
```diff
@@ -2415,37 +2415,40 @@ func TestCombineDuplicateDirectObjects(t *testing.T) {
 			"mollit anim id est laborum.")
 		p.SetTextAlignment(TextAlignmentJustify)
 		p.SetMargins(0, 0, 5, 0)
-		for j := 0; j < 7; j++ {
+
```
Comment (gunnsth) [inline pdf/creator/creator_test.go]: Any reason for the changes in this file?  Isn't it fine to make more objects here for testing the compression?
Also wondering about the page metadata at the bottom, is that relevant to the optimization?

Comment (a5i) [reply pdf/creator/creator_test.go]: I think the file about 1720 KB is not required for this test. After the changes, it has 235 KB.
Add metadata is the easy way to create duplicated direct objects, as I think.
